sotabencheval/core/evaluator.py

```diff
@@ -78,14 +78,39 @@ def cache_exists(self):
                 "No model change detected (using the first batch run "
                 "hash). Will use cached results."
             )
+
             self._cache_exists = True
         else:
             self._cache_exists = False
         return self._cache_exists
 
+    def reset(self):
+        """Resets the internal state of evaluator and allows to start over"""
+        pass
+
     def cache_values(self, **kwargs):
+        """Convert kwargs to something that can be used to generate batch hash"""
         return cache_value(kwargs)
 
+    def eval(self, results_generator):
```

Comment:
I have no idea what the results_generator should look like - if we are to do things this way, then it needs to be documented both in doc strings, and also very clearly in the docs themselves